### PR TITLE
Move `[RCTConvert  RCTTransformOrigin:]` out of UIAccessibilityTraits category

### PR DESCRIPTION
**Pull request by Saadnajmi**
## Summary:

https://github.com/facebook/react-native/commit/5f40f0800e64b4380d7897b2e8b9ff561d84b97c added a new prop and corresponding RCTConvert method for transformOrigin. Strangely, it added the RCTConvert method to a `RCTConvert+UIAccessibilityTraits` category.. which feels like the wrong spot for it. Let's just add it to the existing `RCTConvert+Transform` category. This also means we can get rid of the header if we move the `RCTTransformOrigin` struct into UIView+React (where it is used).

## Changelog:

[IOS][CHANGED] - Move `[RCTConvert  RCTTransformOrigin:]` out of UIAccessibilityTraits category

## Test Plan:

Transform Origin example still works.
Screenshot 2023-10-02 at 11 26 17 AM

### CODE REVIEW

**Review by NickGerleman — APPROVED** [review]
wut

File: packages/react-native/React/Views/UIView+React.h [hunk 1]
```diff
@@ -8,11 +8,17 @@
 #import <UIKit/UIKit.h>
 
 #import <React/RCTComponent.h>
-#import <React/RCTTransformOrigin.h>
+#import <React/RCTConvert.h>
```
Comment (NickGerleman) [inline packages/react-native/React/Views/UIView+React.h:11]: Would it be possible to factor this in a way where it didn't depends on `RCTConvert.h` in the header? This brings in a bunch of other bits like WebKit headers, which makes the build unhappy of some modules which import this file.

File: packages/react-native/React/Views/RCTConvert+Transform.m [hunk 2]
```diff
@@ -144,4 +144,11 @@ + (CATransform3D)CATransform3D:(id)json
   return transform;
 }
 
++ (RCTTransformOrigin)RCTTransformOrigin:(id)json
+{
+  RCTTransformOrigin transformOrigin = {
+      [RCTConvert YGValue:json[0]], [RCTConvert YGValue:json[1]], [RCTConvert CGFloat:json[2]]};
```
Comment (NickGerleman) [inline packages/react-native/React/Views/RCTConvert+Transform.m:150]: For extra credit, remove the usage of Yoga here 😅.

Comment (Saadnajmi) [reply packages/react-native/React/Views/RCTConvert+Transform.m:150]: Like, just convert to a plan CGFloat? sure lol

File: packages/react-native/React/Views/UIView+React.h (hunk 1 above)
Comment (Saadnajmi) [reply packages/react-native/React/Views/UIView+React.h:11]: Hmm, looking at `RCTConvert.h`, it only wants the header for the type `WKDetectorTypes`, of which there is an equivalent `UIDetectorTypes`, that is.. also defined and used. How about I just simplify the logic and only use `UIDetectorTypes`? Ideally.. We shouldn't be depending on Webkit at all, given Hermes exists...

Comment (Saadnajmi) [reply packages/react-native/React/Views/UIView+React.h:11]: If we like that, I can spin up a separate PR, land that, then revisit this one.

Comment (NickGerleman) [reply packages/react-native/React/Views/UIView+React.h:11]: That sounds great!

File: packages/react-native/React/Views/RCTConvert+Transform.m (hunk 2 above)
Comment (NickGerleman) [reply packages/react-native/React/Views/RCTConvert+Transform.m:150]: I think a struct was added for the Fabric version of this, for a value and unit (percentage or dips).

File: packages/react-native/React/Views/UIView+React.h (hunk 1 above)
Comment (Saadnajmi) [reply packages/react-native/React/Views/UIView+React.h:11]: https://github.com/facebook/react-native/pull/39794/
Should build...

Comment (Saadnajmi) [reply packages/react-native/React/Views/UIView+React.h:11]: Neat, I can probably extend this to macOS too lol https://developer.apple.com/documentation/foundation/nsdatadetector

Comment (Saadnajmi) [reply packages/react-native/React/Views/UIView+React.h:11]: er, maybe this one? I dunno blurgh https://developer.apple.com/documentation/appkit/nstextinputtraits/3242733-datadetectiontype

File: packages/react-native/React/Views/RCTConvert+Transform.m (hunk 2 above)
Comment (Saadnajmi) [reply packages/react-native/React/Views/RCTConvert+Transform.m:150]: Merged before I had a chance for this 😅

**Pull request closed without merging**